docs/devGuide/workflow.md

```diff
@@ -125,3 +125,17 @@ ESLint has [integrations with popular editors](https://eslint.org/docs/user-guid
 There are several versions of the _test and linting scripts_ in the main `package.json` that execute for only one of the main or sub packages,
 feel free to look into `package.json` and use them as you see fit!
 </box>
+
+## Updating dependencies
+
+We use `npm install <package folder>` to <tooltip content="take a look under the command in the link!">[hoist](https://docs.npmjs.com/cli/install)</tooltip> the markbind core package's dependencies to the root `node_modules`, avoiding dependency duplication between the `markbind-cli` and `markbind` packages where possible.
+
+Hence, when updating dependencies of the `markbind` core package, be sure to run the `npm install src/lib/markbind` command, or simply the `npm run install:markbind` script. If you upgraded the dependencies of multiple packages, simply run `npm run install:all` instead.
+
+If the dependency <tooltip content="i.e. also listed under the root `package.json`">is also used</tooltip> in the root package, make sure to update it's <tooltip content="as listed in the `package.json` file">version number</tooltip> as well!
+
+<box type="warning">
+
+Conversely, when updating dependencies of the *root* package, be sure to still check whether any subpackages list this as a dependency!
+If so, follow the above process as well.
```

Comment:
> this encourages us to ensure that the core package works

Yup, I think this should be a good direction to move in, and is sort of the second main point of this pr. Otherwise it somewhat defeats the purpose of having the `markbind/markbind-cli` separation. (this also applies if `frontend/components` and `markbind-cli` were to have any duplicated dependencies)

Admiteddly it is a little awkward, which is why I suppose tools like lerna exist (seems like `lerna add` does this for all packages in there) that help keep dependency versions consistent across all packages. I'm not too sure if there is a purely npm solution as well though. 🤔

> Actually, do we still need to list packages in package.json if they are present in src/lib/markbind/package.json?

There's an eslint rule necessitating this, it's also good practice I would think, as it makes clear the dependencies of the main (`cli`) package.

Reply:
> Actually, do we still need to list packages in package.json if they are present in src/lib/markbind/package.json?

But this shouldn't really be applicable once we move all the other non-cli things like `Page.js/Site.js` into `markbind`